src/ECS/Archetypes/AbodeArchetype.cpp

```diff
@@ -49,9 +56,28 @@ entt::entity AbodeArchetype::Create(uint32_t townId, const glm::vec3& position,
 
 	const auto& info = Game::instance()->GetInfoConstants().abode[static_cast<size_t>(type)];
 
-	registry.Assign<Transform>(entity, position, glm::mat3(glm::eulerAngleY(-yAngleRadians)), glm::vec3(scale));
+	const auto& transform =
+	    registry.Assign<Transform>(entity, position, glm::mat3(glm::eulerAngleY(-yAngleRadians)), glm::vec3(scale));
 	registry.Assign<Abode>(entity, info.abodeNumber, townId, foodAmount, woodAmount);
 	registry.Assign<Mesh>(entity, info.meshId, static_cast<int8_t>(0), static_cast<int8_t>(0));
 
+	// Create Fixed component with a 2d bounding circle
+	const auto& bb = game->GetMeshPack().GetMesh(info.meshId).GetBoundingBox();
+	const auto bbSize = glm::max(glm::vec2(1.0f, 1.0f), glm::xz(bb.size() * 0.5f * transform.scale /** transform.scale*/));
+	if (glm::compMax(bbSize) / glm::compMin(bbSize) > 1.4)
+	{
+		throw std::runtime_error("Create non-square bounding circle");
```

Comment:
"NORSE_ABODE_C" throws here

Reply:
In this case, an ellipse should be stored using x, z and angle

Reply:
In this case, a circle is approximated by taking `radius = sqrt(x * x + z * z)` and storing object angle.
I don't think we should store object angle since that should stay in transform and with the box being approximated, the angle shouldn't matter.